### ISSUE TITLE
Random character added to outgoing HTTP Content-Length header

### ISSUE THREAD

**Issue by Fang-** [urbit/urbit#966]
When sending a `%hiss` http request and adding custom headers, an extra character sometimes gets added to the resulting request's Content-Length header.

For example, this hoon code:

```hoon
++  poke-noun
  |=  a=*
  :_  +>.$
  :_  ~
  :-  ost.bol
  :^  %hiss  /some/wire  ~
  :+  %noun  %hiss
  =+  (need (de-purl:html 'http://localhost:8545'))
  :^  -(p.p |)  %post
    ['Content-Type'^['application/json']~ ~ ~]
  (some (as-octt:mimes:html (en-json:html ~)))
```

With `nc -l 8545` running to catch the request, shows:

```
POST / HTTP/1.1
Host: localhost:8545
Content-Type: application/json
9Content-Length: 4

null
```

Note the `9` before the length header. The character is not always a `9`, I've seen an `s` before, and sometimes no character seems to get added at all. I can't get this to happen for requests that don't specify any headers in the `%hiss`.

**Comment by cgyarvin** [comment]
Sounds like a classic nonterminated string bug :-)

**Comment by ohAitch** [comment]
Is this specific to libh2o / pre-h2o cttp?

**Comment by Fang-** [comment]
Did a couple of attempts, can't reproduce on a pre-h2o binary.

**Comment by joemfb** [comment]
Darn it!

(I'm tempted to argue that this wouldn't have been a problem if I'd just stuck with the non-terminated implementation I originally wrote. But that would clearly be a bad argument ...)

**Comment by Fang-** [comment]
Fixed by #967.
